### PR TITLE
feat: --watch mode for extract and compile

### PR DESCRIPTION
**Pull request by jack1913**
https://github.com/lingui/js-lingui/issues/966

I've added 2 functionalities:
1. `--watch` for `lingui extract`
2. `--watch` for `lingui compile` - I think in some implementations of Lingui it can be useful.

### CODE REVIEW

File: packages/cli/src/lingui-extract.ts [hunk 1]
```diff
@@ -51,16 +54,19 @@ export default function command(
...
-    `(use "${chalk.yellow(
-      helpRun("extract")
-    )}" to update catalogs with new messages)`
-  )
-  console.error(
-    `(use "${chalk.yellow(
-      helpRun("compile")
-    )}" to compile catalogs for production)`
-  )
+  if (!options.watch) {
```
Comment (semoal) [inline packages/cli/src/lingui-extract.ts:57]: hm i don't get this logic, why should we log this if watch isn't enabled?

**Review by semoal — CHANGES_REQUESTED** [review]

**Comment by semoal** [comment]
Also, we'll need a new section in the documentation, if you are so busy to work on that I'll find some time next week to introcdee the documentation

Thanks for your work mate

File: packages/cli/src/lingui-compile.ts [hunk 2]
```diff
@@ -153,16 +155,46 @@ if (require.main === module) {
     config.format = program.format
   }
 
-  const results = command(config, {
-    verbose: program.verbose || false,
+  const compile = () => command(config, {
+    verbose: program.watch ? true : program.verbose || false,
```
[the remark this answers is not in the document]
Comment (jack1913) [reply packages/cli/src/lingui-compile.ts:159]: Yeah, you are right. My mistake. Will make an amendment.  Probably today.

File: packages/cli/src/lingui-compile.ts [hunk 3]
```diff
@@ -153,16 +155,46 @@ if (require.main === module) {
...
+    let paths = []; 
+
+    config.locales.forEach((locale) => {
+      catalogs.forEach((catalog) => {
+        paths.push(`${catalog.path.replace(LOCALE, locale).replace(NAME, "*")}.${config.format}`)
+      })  
+    })
+
+    const watcher = chokidar.watch(paths, {
+      persistent: true,
```
[the remark this answers is not in the document]
Comment (jack1913) [reply packages/cli/src/lingui-compile.ts:182]: We do not need here an ignored array, as this watcher only checks locale files in locales catalogs.

File: packages/cli/src/lingui-extract.ts (hunk 1 above)
Comment (jack1913) [reply packages/cli/src/lingui-extract.ts:57]: This is logged always after executing `extract` command. Not my code, not by my assumption. I only added a flag to disable this when the watcher is enabled, as it is illogical to show this in my opinion on extract. But my PR is not about this, so I have left is as is.

**Comment by jack1913** [comment]
@semoal I have updated the docs. I did not make new section, I have updated the CLI section in the REF.

File: packages/cli/src/lingui-compile.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (semoal) [reply packages/cli/src/lingui-compile.ts:159]: If you change this today Jack, we could include this on the tomorrow release

**Review by semoal — CHANGES_REQUESTED** [review]

**Pull request merged**